Tests/GRPCProtobufTests/InputParserTests.swift

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCodeGen
+import GRPCProtobuf
+import SwiftProtobuf
+import SwiftProtobufPluginLibrary
+import XCTest
+
+final class InputParserTests: XCTestCase {
+  func testParser() throws {
+    let parsedCodeGenRequest = try InputParser().parse(input: self.dummyFileDescriptor)
+    XCTAssertEqual(parsedCodeGenRequest.fileName, "helloworld.proto")
+    XCTAssertEqual(
+      parsedCodeGenRequest.leadingTrivia,
+      """
+      // DO NOT EDIT.
+      // swift-format-ignore-file
+      //
+      // Generated by the Swift generator plugin for the protocol buffer compiler.
+      // Source: helloworld.proto
+      //
+      // For information on using the generated types, please see the documentation:
+      //   https://github.com/apple/swift-protobuf/
+
+      // Copyright 2015 gRPC authors.\n//\n// Licensed under the Apache License, \
+      Version 2.0 (the \"License\");\n// you may not use this file except in \
+      compliance with the License.\n// You may obtain a copy of the License \
+      at\n//\n//     http://www.apache.org/licenses/LICENSE-2.0\n//\n// Unless required by \
+      applicable law or agreed to in writing, software\n// distributed under \
+      the License is distributed on an \"AS IS\" BASIS,\n// WITHOUT WARRANTIES \
+      OR CONDITIONS OF ANY KIND, either express or implied.\n// See the License \
+      for the specific language governing permissions and\n// limitations under the License.\n\n
```

Comment:
I think this somewhat misses the point of multiline string literals..., you don't need to put in "\n" manually, you just write the text as you'd like it. The `\` at the end of each line _stops_ the newline being inserted.

See https://docs.swift.org/swift-book/documentation/the-swift-programming-language/stringsandcharacters#Multiline-String-Literals